pkg/apis/core/types.go

```diff
@@ -4255,11 +4255,44 @@ type NodeAddressType string
 
 // These are valid values of node address type
 const (
-	NodeHostName    NodeAddressType = "Hostname"
-	NodeExternalIP  NodeAddressType = "ExternalIP"
-	NodeInternalIP  NodeAddressType = "InternalIP"
-	NodeExternalDNS NodeAddressType = "ExternalDNS"
+	// NodeHostName identifies a name of the node. Although every node can be assumed
+	// to have a NodeAddress of this type, its exact syntax and semantics are not
+	// defined, and are not consistent between different clusters.
```

Comment:
Non-blocking comment: we probably want to mention here that this usually matches the hostname according to the OS, although it's not strictly required.

Reply:
Per https://github.com/kubernetes/kubernetes/pull/95768#discussion_r612656206 I don't think that's true

Reply:
I did say "usually" ;) "primary DNS name" and "hostname" is typically the same from my experience